### PR TITLE
Work in progress

### PR DESCRIPTION
**Pull request by mmarganski**

### CODE REVIEW

File: src/modules/app/app.service.ts [hunk 1]
```diff
@@ -57,21 +59,48 @@ export class AppService {
...
         return this.roomRepository.save({
             roomName,
-            isPrivate
+            isPrivate,
+            messages: []
         })
     }
 
-    createMessage (messageContent: string, isImage: boolean, date: Date, userId: string, roomName: string) {
+    async createMessage (messageContent: string, isImage: boolean, date: Date, userId: string, roomName: string) {
```
Comment (WadasGrzegorz) [inline src/modules/app/app.service.ts]: if you have so many parameters you should create one parameter like `message: CreateChatMessage`

File: src/modules/app/app.service.ts [hunk 2]
```diff
@@ -57,21 +59,48 @@ export class AppService {
...
+            roomName,
+            socketId,
+            createdAt: new Date(),
+            updatedAt: new Date()
         })
     }
 
+    async addUserToRoom (roomName: string, socketId: string) {
+        const userRoom = await this.userRoomRepository.findOne({ where: { roomName, socketId } })
+        if (userRoom === undefined) {
```
Comment (WadasGrzegorz) [inline src/modules/app/app.service.ts]: `if (!userRoom)`

Comment (WadasGrzegorz) [inline src/modules/app/app.service.ts]: enter

File: package.json [hunk 3]
```diff
@@ -18,7 +18,9 @@
     "test:watch": "jest --watch",
     "test:cov": "jest --coverage",
     "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
-    "test:e2e": "jest --config ./test/jest-e2e.json"
+    "test:e2e": "jest --config ./test/jest-e2e.json",
+    "prepare": "rd /s /q .husky && husky install && npx husky add .husky/pre-commit \"yarn lint\" ",
```
Comment (WadasGrzegorz) [inline package.json]: use rimraf
husky add without npx

File: src/modules/app/app.service.ts [hunk 4]
```diff
@@ -57,21 +59,48 @@ export class AppService {
...
             roomName,
-            isPrivate
+            isPrivate,
+            messages: []
         })
     }
 
-    createMessage (messageContent: string, isImage: boolean, date: Date, userId: string, roomName: string) {
+    async createMessage (messageContent: string, isImage: boolean, date: Date, userId: string, roomName: string) {
+        const currentRoom = await this.roomRepository.findOne({where: { roomName } })
```
Comment (WadasGrzegorz) [inline src/modules/app/app.service.ts]: space before where

File: src/modules/app/app.service.ts [hunk 5]
```diff
@@ -57,21 +59,48 @@ export class AppService {
...
+        }
+    }
+
     deactivateUser (socketId: string) {
         return this.userRepository.update({ socketId },{ isActive: false })
     }
+
+    async getUsersInRoom (roomName: string) {
+        return createQueryBuilder('UserEntity')
+            .leftJoinAndSelect(UserRoomEntity, 'UserRoomEntity', 'UserEntity.socketId = UserRoomEntity.socketId')
```
Comment (WadasGrzegorz) [inline src/modules/app/app.service.ts]: use alias like 'UR'

**Review by WadasGrzegorz — CHANGES_REQUESTED** [review]

File: src/modules/app/app.service.ts [hunk 6]
```diff
@@ -57,21 +59,48 @@ export class AppService {
...
+        return this.userRoomRepository.save({
+            roomName,
+            socketId,
+            createdAt: new Date(),
+            updatedAt: new Date()
         })
     }
 
+    async addUserToRoom (roomName: string, socketId: string) {
+        const userRoom = await this.userRoomRepository.findOne({ where: { roomName, socketId } })
```
Comment (jpudysz) [inline src/modules/app/app.service.ts:104]: enter between consts and ifs. Group assignments together and instructions together

**Review by jpudysz — CHANGES_REQUESTED** [review]

**Review by jpudysz — APPROVED** [review]

File: src/lib/entities/message.entity.ts [hunk 7]
```diff
@@ -7,10 +7,10 @@ export class MessageEntity {
     messageId: number
 
     @Column()
-    messageContent: string
+    messageText: string
 
-    @Column()
-    isImage: boolean
+    @Column('longblob')
```
Comment (WadasGrzegorz) [inline src/lib/entities/message.entity.ts:12]: we don't save whole image in db instead use imageUrl and save file on disk

File: src/lib/entities/user.entity.ts [hunk 8]
```diff
@@ -8,7 +8,7 @@ export class UserEntity {
     @Column()
     userName: string
 
-    @Column()
+    @Column('longblob')
```
Comment (WadasGrzegorz) [inline src/lib/entities/user.entity.ts:11]: aa

File: src/types.ts [hunk 9]
```diff
@@ -17,3 +17,10 @@ export type MessageCall = {
 export enum Rooms {
     All = 'All'
 }
+
+export type CreateChatMessage = {
+    messageContent: string,
+    isImage: boolean,
+    socketId: string,
+    roomName: string
+}
```
Comment (WadasGrzegorz) [inline src/types.ts]: nl

File: src/types.ts [hunk 10]
```diff
@@ -17,3 +17,10 @@ export type MessageCall = {
 export enum Rooms {
     All = 'All'
 }
+
+export type CreateChatMessage = {
```
Comment (WadasGrzegorz) [inline src/types.ts:21]: if you have common types please move it to `lib/types` for example `common.ts` or `chat.ts`

**Review by WadasGrzegorz — CHANGES_REQUESTED** [review]

**Review by WadasGrzegorz — APPROVED** [review]

**Pull request merged**